scripts/firedrake-install

```diff
@@ -100,6 +118,53 @@ jenkins = "JENKINS_URL" in os.environ
 ci_testing_firedrake = "FIREDRAKE_CI_TESTS" in os.environ
 
 
+def resolve_doi_branches(doi):
+    import requests
+    import hashlib
+    log.info("Installing Firedrake components specified by {}".format(doi))
+    response = requests.get("https://zenodo.org/api/records",
+                            params={"q": "doi:{}".format(doi.replace("/", r"\/"))})
+    if response.status_code >= 400:
+        log.error("Unable to obtain Zenodo record for doi {}".format(doi))
+        log.error("Response was {}".format(response.json()))
+        sys.exit(1)
+    response = response.json()
+    try:
+        record, = response["hits"]["hits"]
+    except ValueError:
+        log.error("Was expecting one record for doi '{doi}', found {num}".format(
+            doi=doi, num=response["hits"]["total"]))
+        log.error("Response was {}".format(response))
+        sys.exit(1)
+    files = record["files"]
+    try:
+        componentjson, = (f for f in files if f["key"] == "components.json")
+    except ValueError:
+        log.error("Expecting to find exactly one 'components.json' in record")
+        sys.exit(1)
+    download = requests.get(componentjson["links"]["self"])
+    if download.status_code >= 400:
+        log.error("Unable to download 'components.json'")
+        log.error("Response was {}".format(download.json()))
+        sys.exit(1)
+    # component response has checksum as "md5:HEXDIGEST", strip the md5.
+    if hashlib.md5(download.content).hexdigest() != componentjson["checksum"][4:]:
+        log.error("Download failed checksum, expecting {expect}, got {got}".format(
+            expect=componentjson["checksum"][4:],
+            got=hashlib.md5(download.content).hexdigest()))
+        sys.exit(1)
+    componentjson = download.json()
+    branches = {}
+    for record in componentjson["components"]:
+        commit = record["commit"]
```

Comment:
I use the commit, rather than the tag, because the latter might have been moved.